src/crypto/certificate_pool.rs

```diff
@@ -66,14 +99,16 @@ impl CertificatePool {
         })?;
         let cert = picky::x509::Cert::from_pem_str(&cert_pem_str)?;
 
-        let verified = self.trusted_roots.iter().any(|trusted_root| {
-            let chain = [trusted_root.clone()];
-            cert.verifier()
-                .chain(chain.iter())
-                .exact_date(&cert.valid_not_before())
-                .verify()
-                .is_ok()
-        });
+        let verified = self
+            .create_chains_for_all_certificates()
+            .iter()
+            .any(|chain| {
+                cert.verifier()
+                    .chain(chain.iter().cloned())
```

Comment:
`chain` is a `Vec<&Cert>`, when we call `iter` on it we end up iterating over `&&cert`.

Instead of cloning them, we could do something like:

```rust
.chain(chain.iter().map(|c| *c))
```

which is equivalent to:

```suggestion
                    .chain(chain.iter().copied())
```

I think this works and should be faster than clone. Can you double check please?